sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -1722,6 +1723,46 @@ def _get_runbook_path_from_agent_spec(
             result=runbook_path,
         ).as_dict()
 
+    @command_dispatcher(commands.SEMA4AI_REFRESH_AGENT_SPEC_INTERNAL)
+    def _refresh_agent_spec(self, params: RefreshAgentSpecDict) -> ActionResult:
+        action_spec_path = Path(params["action_spec_path"])
+
+        from ruamel.yaml import YAML
+
+        yaml = YAML()
+        yaml.preserve_quotes = True
+
+        with action_spec_path.open("r") as file:
+            package_yaml = yaml.load(file.read())
+
+        agent_spec_path = Path(
+            *action_spec_path.parts[: action_spec_path.parts.index("actions")],
+            "agent-spec.yaml",
+        )
+
+        with agent_spec_path.open("r") as file:
+            agent_spec_yaml = yaml.load(file.read())
+
+        relative_action_path = Path(
+            *action_spec_path.parts[action_spec_path.parts.index("actions") + 1 : -1]
+        )
+
+        agent_spec_yaml["agent-package"]["agents"][0]["action-packages"].append(
+            {
+                "name": package_yaml["name"],
+                "organization": "MyActions",
```

Comment:
Actually, the organization name should be based on the folder name under `actions` where the action was found (and if it was `Sema4.ai` it should be skipped -- probably should check all lowercase and removing the `.` when doing the compare so that it compares to `sema4ai`).

Reply:
To align the agent content at least a bit I would really like to limit the custom actions to by only under MyActions and not allow the users to do "whatever". So if the actions are not under MyActions or Sema4ai they just are not valid.
This structure will be enforced in any case in Studio and Agent Server, which would meant that if you import and agent to Studio and then export it back out the folders will change.. so atm better to enforce here as well.

(NOTE: This is a thing I think might be the main driver of agentSpec v3 in the near future)